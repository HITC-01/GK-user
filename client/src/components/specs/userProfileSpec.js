import React from 'react';
import renderer from 'react-test-renderer';
import UserProfile from '../userProfile.jsx';
import { shallow } from '../../enzyme';


describe('<UserProfile />', () => {
  let props = {};
  
  beforeEach(()=>{
    props = {
      id: 2,
      pro: 'true',
      isFollowing: 'false',
      followers: 69,
      trackCount: 28,
      userName: 'Trevor Ankunding IV',
      profilePhoto: 'https://s3.amazonaws.com/uifaces/faces/twitter/jjshaw14/128.jpg',
      location: 'Davismouth, Montserrat',
    };
  });
    
  test('Should render correctly if no userData is passed', () => {
    const component = renderer.create(<UserProfile />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });


  test('Should correctly render userdata', () => {
    const component = renderer.create(<UserProfile userData={props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render following state', () => {
    const component = renderer.create(<UserProfile following={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should work with props', () => {
    const component = shallow(<UserProfile {...props} />);
    const propsOut = Array.from(Object.keys(component.props()));
    expect(propsOut.length).toBe(2);
  });
});
