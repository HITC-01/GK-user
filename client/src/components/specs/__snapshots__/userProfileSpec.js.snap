// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> Should render correctly if no userData is passed 1`] = `
<div
  className="up-container"
>
  <img
    alt="Loading"
    className="up-photo"
  />
  <br />
  <span
    className="up-username"
    title="Visit undefined's profile"
  >
    
  </span>
  <br />
  <i
    className="fas fa-users"
  />
  <span
    className="up-followcount black"
    title="undefined followers"
  />
  <i
    className="fas fa-headphones-alt"
  />
  <span
    className="up-trackcount black"
    title="undefined tracks"
  />
  <br />
  <button
    className="up-button up-follow"
    onClick={[Function]}
    title="follow"
    type="button"
  >
    follow
  </button>
  <br />
  <div
    id="up-report-sign"
  >
    <i
      className="far fa-flag"
    />
    <button
      className="up-report-button black"
      onClick={[Function]}
      type="button"
    >
      Report
    </button>
  </div>
</div>
`;
